backend/controllers/post_controller.py

```diff
@@ -32,3 +37,122 @@ def get_posts(c_id: str) -> List[Post]:
     except Exception as e:
         print(f"Error in get_posts: {type(e).__name__}: {str(e)}")
         raise HTTPException(status_code=500, detail=f"Failed to fetch posts: {str(e)}")
+
+
+def get_post(c_id: str, post_id: str) -> Post:
+    try:
+        with get_db() as db:
+            post = (
+                db.query(Post).filter(Post.course_id == c_id, Post.id == post_id).first()
+            )
+            return post
+    except Exception as e:
+        print(f"Error in get_post: {type(e).__name__}: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to fetch posts: {str(e)}")
+
+
+def update_post(c_id: str, user_id: str, post_edit_info: CreatePostEditRequest) -> PostEditResponse:
+    try:
+        with get_db() as db:
+            post_id = post_edit_info.post_id
+            post = (
+                db.query(Post).filter(Post.course_id == c_id, Post.id == post_id).first()
+            )
+
+            if not post:
+                raise HTTPException(status_code=404, detail="Post not found")
+
+            post.content = post_edit_info.new_content
+
+            post_edit = PostEdit(
+                post_id=post_edit_info.post_id,
+                edited_by=user_id,
+                new_content=post_edit_info.new_content,
+                edit_reason=post_edit_info.edit_reason,
+            )
+
+            db.add(post_edit)
+            db.flush()
+            id = UUID(str(post_edit.id))
+            return PostEditResponse(
+                id=id,
+                post_id=post_edit_info.post_id,
+                edited_by=UUID(user_id),
+                new_content=post_edit_info.new_content,
+                edit_reason=post_edit_info.edit_reason,
+            )
+
+    except Exception as e:
+        print(f"Error in update_post: {type(e).__name__}: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to fetch posts: {str(e)}")
```

Comment:
slight issue with this is that the 500 exception catches all specific errors and raises 500. You want the 404 persist if that is the issue. a possible fix:

```python
except HTTPException as e:
    # Re-raise HTTPErrors
    # Important so that 404 from "Post not found" isn't caught by the generic Exception block
    raise

except Exception as e:
    # Catch any other unexpected exception
    print(f"Unexpected error in update_post: {type(e).__name__}: {str(e)}")
    raise HTTPException(status_code=500, detail=f"An unexpected error occurred: {str(e)}")
    ```